src/cogent3/app/composable.py

```diff
@@ -1019,6 +1022,8 @@ def wrapped(klass):
         if hasattr(klass, "__slots__"):
             klass.__slots__ += tuple(slot_attrs)
 
+        # register this app
+        __app_registry[get_object_provenance(klass)] = klass.__name__
```

Comment:
I would write this as:
__app_registry[klass.__name__] = get_object_provenance(klass)

Reply:
I understand this logic, but why a user may want to have two classes with the same name.

Reply:
I think the issue is more ensuring registration of the built-in apps is not overridden by another developer. It also means it's easier to check whether an object is composable by looking at full provenance.